src/components/services/Services.tsx

```diff
@@ -55,6 +60,37 @@ export function Services(props: Props) {
     dispatch(checkCompromisedKeys(navigation)).catch(e => {
       console.warn('checkCompromisedKeys error:', e)
     })
+
+    // FIO handle registration:
+    const { freeRegApiToken = '', freeRegRefCode = '' } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
+    const fioWalletsWithAddresses = (
+      await Promise.all(
+        Object.values(account.currencyWallets).reduce<Array<Promise<EdgeCurrencyWallet>>>((promises, wallet) => {
+          if (wallet.currencyInfo.pluginId === 'fio') {
+            promises.push(
+              wallet.otherMethods.getFioAddressNames().then((fioNames: string[]) => {
+                if (fioNames.length > 0) {
+                  return wallet
+                }
+              })
+            )
+          }
+          return promises
+        }, [])
+      )
+    ).filter(wallet => wallet != null)
+
+    if (fioWalletsWithAddresses.length === 0 && freeRegApiToken !== '' && freeRegRefCode !== '') {
+      const itemIds = await account.dataStore.listItemIds('')
+      if (!itemIds.includes(FIO_CREATE_HANDLE_IGNORED_AT)) {
+        const isCreateHandle = await Airship.show<boolean>(bridge => <FioCreateHandleModal bridge={bridge} />)
```

Comment:
Maybe call this "shouldCreate". We agreed not to start booleans with "is", but instead to give them sensible names.